### PR TITLE
Added Licence field to bulk-edit form on Backoffice Asset.

### PR DESCRIPTION
**Pull request by ghost**
Adding licence field to asset's bulk edit form is hard using regualar Django
mechanism. This pull request fixes that.

### CODE REVIEW

File: src/ralph/back_office/admin.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ class BackOfficeAssetAdmin(
     ]
     resource_class = resources.BackOfficeAssetResource
     bulk_edit_list = [
-        'status', 'barcode', 'imei', 'hostname', 'model', 'purchase_order',
+        'licence', 'status', 'barcode', 'imei', 'hostname', 'model', 'purchase_order',
```
Comment (mkurek) [inline src/ralph/back_office/admin.py]: licences

File: src/ralph/back_office/admin.py [hunk 2]
```diff
@@ -151,6 +151,67 @@ class BackOfficeAssetAdmin(
...
+        #TODO:: this doesn't check permissions?
+        #TODO:: only if bulk
+        from django import forms
+        from ralph.licences.models import Licence
+        from ralph.admin.sites import ralph_site
+        from ralph.admin.widgets import AutocompleteWidget
+        from django.db.models.loading import get_model
+        Form = super().get_changelist_form(request, **kwargs)
+        class NewForm(Form):
+            licence = forms.ModelMultipleChoiceField(
```
Comment (mkurek) [inline src/ralph/back_office/admin.py]: licences

File: src/ralph/back_office/admin.py [hunk 3]
```diff
@@ -151,6 +151,67 @@ class BackOfficeAssetAdmin(
...
+                    kwargs['instance'].licences.values_list('licence__id', flat=True) # noqa
+                ]
+                kwargs['initial'] = initial
+                super().__init__(*args, **kwargs)
+
+            def save_m2m(self, commit=True):
+                pass
+
+            def save(self, commit=True):
+                form_licences = self.cleaned_data['licence']
```
Comment (mkurek) [inline src/ralph/back_office/admin.py]: move this logic to `save_m2m` (and call super there)

**Comment by mkurek** [comment]
Could you leave some `TODO: permissions handling` in your code?

**Comment by ghost** [comment]
done

**Comment by mkurek** [comment]
:ok:

**Comment by szok** [comment]
:ok:

**Pull request merged**